File: Logitude/Simplog.Data/Helpers/AutomationOnUpdateDocument.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace Simplog.Data.Helpers
{

    [DataContract(Namespace = "")]
    public class AutomationOnUpdateDocument
    {
        [DataMember]
        public string SendVia { get; set; }
        [DataMember]
        public FTPAutomationDetails FTPDetails { get; set; }
        [DataMember]
        public string ComputingPartnerId { get; set; }
        [DataMember]
        public List<OnUpdateDocumentTypeAttachment> DocumentTypeLists { get; set; }
    }

    [DataContract(Namespace = "")]
    public class OnUpdateDocumentTypeAttachment
    {
        [DataMember]
        public string DocumentTypeId { get; set; }
        [DataMember]
        public string DocumentTypeName { get; set; }
        [DataMember]
        public string DocumentTypeCopyId { get; set; }
        [DataMember]
        public string Type { get; set; }
    }
}

File: Logitude/AngularModules/AngularModules/Infrastructure/DataContracts/AutomationOnUpdateDocument.ts
import { OnUpdateDocumentTypeAttachment } from '../Components/Maintenance/Automation/DocumentAttachmentsComponent';

export class AutomationOnUpdateDocument {

    constructor() {
        this.FTPDetails = new FTPAutomationDetails();
    }

    public SendVia: string;
    public FTPDetails: FTPAutomationDetails;
    public ComputingPartnerId: string;
    public DocumentTypeLists: OnUpdateDocumentTypeAttachment[];
    public IsChanged: boolean;
}

export class FTPAutomationDetails {
    public Host: string;
    public Folder: string;
    public UserName: string;
    public Password: string;
}
